### PR TITLE
[FLINK-19868][csv] Csv Serialization schema contains line delimiter

### PR DESCRIPTION
**Pull request by SteNicholas**
## What is the purpose of the change

*`CsvRowSerializationSchema` and `CsvRowDataSerializationSchema` serialize `Row` and `RowData` to byte array end with `\n` at default. For the row-wise formats, it should consume and produce the bytes for a row.  `csv.line-delimiter` option is not a good option for csv format and shoud be removed for `CsvRowSerializationSchema` and `CsvRowDataSerializationSchema` to not append a new line.*

## Brief change log

  - *`CsvSchema` in `CsvRowSerializationSchema` and `CsvRowDataSerializationSchema` configures the line separator to empty string.*
  - *`line-delimiter` option in `CsvFormatFactory`, `CsvRowFormatFactory` and `CsvFileSystemFormatFactory` is removed.*
  - *`CsvFileSystemFormatFactory` encodes `RowData` record end with line delimiter `\n` to write.

## Verifying this change

  - *Remove the line delimiter config of test cases in `CsvRowDataSerDeSchemaTest` and `CsvRowDeSerializationSchemaTest` to verify whether the byte array which `CsvRowSerializationSchema` and `CsvRowDataSerializationSchema` serialize `Row` and `RowData` to includes line delimiter `\n`.*
  - *Remove the`line-delimiter` option of test cases in `CsvFormatFactoryTest` and `CsvRowFormatFactoryTest` to verify whether the `line-delimiter` option could be removed normally.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by SteNicholas** [comment]
@wuchong , could you please help to review this pull request if you are available?

File: docs/dev/table/connectors/formats/csv.md [hunk 1]
```diff
@@ -95,18 +95,6 @@ Format Options
...
-    <tr>
-      <td><h5>csv.line-delimiter</h5></td>
-      <td>optional</td>
-      <td style="word-wrap: break-word;"><code>\n</code></td>
-      <td>String</td>
-      <td>Line delimiter, <code>\n</code> by default. Note the <code>\n</code> and <code>\r</code> are invisible special characters, you have to use unicode to specify them in plain SQL.
-          <ul>
-           <li>e.g. <code>'csv.line-delimiter' = U&'\000D'</code> specifies the to use carriage return <code>\r</code> as line delimiter.</li>
-           <li>e.g. <code>'csv.line-delimiter' = U&'\000A'</code> specifies the to use line feed <code>\n</code> as line delimiter.</li>
-          </ul>
```
Comment (wuchong) [inline docs/dev/table/connectors/formats/csv.md:107]: Move this notes to {{field-delimiter}} option.

File: flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvFileSystemFormatFactory.java [hunk 2]
```diff
@@ -194,7 +186,12 @@ private CsvSchema buildCsvSchema(RowType rowType, ReadableConfig options) {
 
 		final CsvRowDataSerializationSchema serializationSchema = builder.build();
 
-		return Optional.of((record, stream) -> stream.write(serializationSchema.serialize(record)));
+		return Optional.of((record, stream) -> {
+			StringBuilder sb = new StringBuilder();
+			sb.append(new String(serializationSchema.serialize(record)));
+			sb.append("\n");
+			stream.write(sb.toString().getBytes());
```
Comment (wuchong) [inline flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvFileSystemFormatFactory.java:193]: Why we hard code `\n` ? I think the purpose of this issue is removing the "\n".

File: flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvRowFormatFactory.java [hunk 3]
```diff
@@ -101,9 +100,6 @@ public CsvRowFormatFactory() {
 		descriptorProperties.getOptionalCharacter(CsvValidator.FORMAT_FIELD_DELIMITER)
 			.ifPresent(schemaBuilder::setFieldDelimiter);
 
-		descriptorProperties.getOptionalString(CsvValidator.FORMAT_LINE_DELIMITER)
-			.ifPresent(schemaBuilder::setLineDelimiter);
```
Comment (wuchong) [inline flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvRowFormatFactory.java:105]: I would recommend not changing the legacy csv format.

File: flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvRowSerializationSchema.java [hunk 4]
```diff
@@ -86,8 +86,8 @@ private CsvRowSerializationSchema(
 		this.typeInfo = typeInfo;
 		this.runtimeConverter = createRowRuntimeConverter(typeInfo, true);
 		this.csvMapper = new CsvMapper();
-		this.csvSchema = csvSchema;
-		this.objectWriter = csvMapper.writer(csvSchema);
+		this.csvSchema = csvSchema.withLineSeparator("");
```
Comment (wuchong) [inline flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvRowSerializationSchema.java:89]: ditto. I would recommend not changing the legacy csv format.

**Review by wuchong — APPROVED** [review]
LGMT.

Waiting for the build to pass.

**Comment by wuchong** [comment]
The failed e2e test seems not related to this PR. 

Will merge it.

**Pull request merged**